browser/components/newtab/aboutNewTabService.js

```diff
@@ -128,10 +138,14 @@ AboutNewTabService.prototype = {
       NewTabPrefsProvider.prefs.on(
         PREF_MATCH_OS_LOCALE,
         this._updateRemoteMaybe.bind(this));
+      NewTabPrefsProvider.prefs.on(
+        PREF_REMOTE_MODE,
+        this._updateRemoteMaybe.bind(this));
       this._remoteEnabled = true;
     } else {
       NewTabPrefsProvider.prefs.off(PREF_SELECTED_LOCALE, this._updateRemoteMaybe);
```

Comment:
I've found out that `off` may not work due to `this` binding. This could be a good opportunity to fix that bug.
In the init function, can you please see if `this` is bound appropriately?

e.g.

```javascript
function AboutNewTabService() {
  NewTabPrefsProvider.prefs.on(PREF_REMOTE_ENABLED, this._handleToggleEvent.bind(this));
  this._updateRemoteMaybe = this._updateRemoteMaybe.bind(this);
  ...
}
```

Can you please verify if `off` won't work in the case `on` has been set with `this._updateRemoteMaybe.bind(this)`?
And if so, would the solution cited above work?